packages/data-service-generator/src/admin/create-dotenv.ts

```diff
@@ -1,25 +1,87 @@
-import { AppInfo, Module } from "@amplication/code-gen-types";
+import {
+  CreateAdminDotEnvParams,
+  EventNames,
+  Module,
+  ModuleMap,
+  VariableDictionary,
+} from "@amplication/code-gen-types";
 import { readCode } from "@amplication/code-gen-utils";
 import { replacePlaceholdersInCode } from "../utils/text-file-parser";
+import pluginWrapper from "../plugin-wrapper";
+import DsgContext from "../dsg-context";
 
 const templatePath = require.resolve("./create-dotenv.template.env");
 
+export async function createDotEnvModule(
+  envVariables: VariableDictionary = []
+): Promise<ModuleMap> {
+  return pluginWrapper(
+    createDotEnvModuleInternal,
+    EventNames.CreateAdminDotEnv,
+    {
+      envVariables,
+    }
+  );
+}
+
 /**
  * Creates the .env file based on the given template with placeholder.
  * The function replaces any placeholder in a ${name} format based on the key in the appInfo.settings
- * @returns grants JSON module
+ * @returns grants JSON modulemaps
  */
-export async function createDotEnvModule(
-  appInfo: AppInfo,
-  baseDirectory: string
-): Promise<Module> {
-  const MODULE_PATH = `${baseDirectory}/.env`;
+export async function createDotEnvModuleInternal({
+  envVariables,
+}: CreateAdminDotEnvParams): Promise<ModuleMap> {
+  const context = DsgContext.getInstance;
+  const { appInfo, clientDirectories } = context;
+
+  const MODULE_PATH = `${clientDirectories.baseDirectory}/.env`;
   const code = await readCode(templatePath);
+  const envVariablesWithoutDuplicateKeys = removeDuplicateKeys(envVariables);
+  const formattedAdditionalVariables = convertToKeyValueSting(
```

Comment:
@Shurtu-gal I think this is a typo `convertToKeyValueSting` instead of `convertToKeyValueString` (missing 'r' on 'String')

Reply:
Done